src/snowflake/snowpark/mock/_functions.py

```diff
@@ -1694,6 +1694,44 @@ def mock_date_trunc(part: str, datetime_expr: ColumnEmulator) -> ColumnEmulator:
     return ColumnEmulator(truncated, sf_type=datetime_expr.sf_type)
 
 
+@patch("datediff")
+def mock_datediff(
+    part: str, col1: ColumnEmulator, col2: ColumnEmulator
```

Comment:
are col1 and col2 guaranteed to be of type timestamp -- is it possible that col1 is of string type while col2 is int type?

Reply:
In sql both columns have to be either dates or timestamps. I could add a validation for column type, but I think we've avoided doing that for other functions so far.